omniledger/contracts/coins.go

```diff
@@ -123,35 +124,37 @@ func ContractCoin(cdb ol.CollectionView, inst ol.Instruction, c []ol.Coin) (sc [
 				return
 			}
 
-			targetCoin := newSafeUint64(v)
-			targetCoin, err = targetCoin.add(coinsArg)
+			var targetCI CoinInstance
+			err = protobuf.Decode(v, &targetCI)
+			if err != nil {
+				return nil, nil, errors.New("couldn't unmarshal target account: " + err.Error())
+			}
+			targetCI.Balance, err = safeAdd(targetCI.Balance, coinsArg)
```

Comment:
I wonder if safeAdd should instead be a method on CoinInstance: `func (ci *CoinInstance)add(x int) error`?